fedimg/services/ec2/ec2imguploader.py

```diff
@@ -0,0 +1,386 @@
+                    architecture=self.image_architecture,
+                    block_device_mapping=block_device_map,
+                    root_device_name=root_device_name
+                )
+
+                if self.push_notifications:
+                    fedimg.messenger.notify(
+                        topic='image.upload',
+                        msg=dict(
+                            image_url=self.image_url,
+                            image_name=self.image_name,
+                            destination=self.region,
+                            service=self.service,
+                            status='completed',
+                            compose=self.compose_id,
+                            extra=dict(
+                                id=image.id,
+                                virt_type=self.image_virtualization_type,
+                                vol_type=self.image_volume_type
+                            )
+                        )
+                    )
+
+                return image
+
+            except Exception as e:
+                log.info('Could not register with name: %r' % self.image_name)
+                if 'InvalidAMIName.Duplicate' in str(e):
+                    counter = counter + 1
+                else:
+                    raise
+
+    def _remove_volume(self, volume):
+        log.info('[CLEAN] Destroying volume: %r' % volume.id)
+        self._connect().destroy_volume(volume)
+
+    def clean_up(self, image_id, delete_snapshot=True, force=False):
+        """
+        Clean the resources created during the upload process
+
+        Args:
+            image_id (str): ID of the AMI
+            delete_snapshot (bool): Bool to control snapshot deletion along
+                                    with AMI.
+            force: Override AWS_DELETE_RESOURCES config value
+
+        Returns:
+            Boolean: True, if resources are deleted else False
+        """
+        if not AWS_DELETE_RESOURCES and force:
+            log.info('Deleting resource is disabled by config.'
+                     'Override by passing force=True.')
+            return False
+
+        if self.volume:
+            self._connect().destroy_volume(self.volume)
+
+        self._connect().deregister_image(
+            image_id,
+            delete_snapshot=delete_snapshot
+        )
+
+        return True
+
+    def set_image_virt_type(self, virt_type):
+        """
+        Set the `image_virtualization_type` attribute of the `EC2ImageUploader`
+        object.
+
+        Args:
+            virt_type (str): virtualization type to set for the object.
+        """
+        self.image_virtualization_type = virt_type
+
+    def set_image_url(self, image_url):
+        """
+        Set the `image_url` attribute of the `EC2ImageUploader` object.
+
+        Args:
+            image_url (str): image_url to set for the object.
+        """
+        self.image_url = image_url
```

Comment:
This type of setter while valid aren't really pythonic, most people would user: ``object.attribute = value``